lib/virtfs/nativefs/thick/dir_class_methods.rb

```diff
@@ -5,30 +5,30 @@ module VirtFS::NativeFS # rubocop:disable Style/ClassAndModuleChildren
   class Thick
     module DirClassMethods
       def dir_delete(p)
-        VfsRealDir.delete(p)
+        VfsRealDir.delete(apply_root(p))
       end
 
       def dir_entries(p)
-        VfsRealDir.entries(p)
+        VfsRealDir.entries(apply_root(p))
       end
 
       def dir_exist?(p)
-        VfsRealDir.exist?(p)
+        VfsRealDir.exist?(apply_root(p))
       end
 
       def dir_foreach(p, &block)
-        VfsRealDir.foreach(p, &block)
+        VfsRealDir.foreach(apply_root(p), &block)
       end
 
       def dir_mkdir(p, permissions)
-        VfsRealDir.mkdir(p, permissions)
+        VfsRealDir.mkdir(apply_root(p), permissions)
       end
 
-      def dir_new(_fs_rel_path, hash_args, open_path, cwd)
+      def dir_new(fs_rel_path, hash_args, _open_path, cwd)
         owd = VfsRealDir.getwd
         begin
           VfsRealDir.chdir(cwd)
-          return VfsRealDir.new(open_path, hash_args)
+          return VfsRealDir.new(apply_root(fs_rel_path), hash_args)
```

Comment:
Yep, it's intentional. Going forward I'll have to re-evaluate the need for `open_path`.